core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -862,6 +854,15 @@ private[spark] class ExecutorAllocationManager(
       }
     }
 
+    override def onExecutorAllocatorRangeUpdate(
+        event: SparkListenerExecutorAllocatorRangeUpdate): Unit = {
+      val newLower = event.lower.getOrElse(_minNumExecutors)
+      val newUpper = event.upper.getOrElse(_maxNumExecutors)
+      validateSettings(newLower, newUpper)
```

Comment:
I believe this should be like
```
try {
   validateSettings(newLower, newUpper)
   _minNumExecutors = newLower
     _maxNumExecutors = newUpper
} catch {
 ....
}
```
Otherwise exception is thrown in this listener.